DatadogSessionReplay/Sources/Recorder/ViewTreeSnapshotProducer/ViewTreeSnapshot/NodeIDGenerator.swift

```diff
@@ -32,28 +33,32 @@ internal final class NodeIDGenerator {
 
     /// Returns single `NodeID` for given instance of `UIView`.
     /// - Parameter view: the `UIView` object
+    /// - Parameter nodeRecorder: the `NodeRecorder` object
     /// - Returns: the `NodeID` of queried instance
-    func nodeID(for view: UIView) -> NodeID {
-        if let currentID = view.nodeID {
-            return currentID
+    func nodeID(view: UIView, nodeRecorder: NodeRecorder) -> NodeID {
+        let key = nodeRecorder.identifier.uuidString + "-\(view.hash)"
```

Comment:
How does `view.hash` work exactly? I can only find its definition on [`NSObjectProtocol.hash`](https://developer.apple.com/documentation/objectivec/nsobjectprotocol/1418859-hash) but nothing in there assures me that we can use it reliably for `UIView` tagging. In other words, are we sure the `hash` will be stable for `UIView` instances reused among view controllers or cells in recyclable lists?

Reply:
Hash is an address in memory. We already use it for image recording.

It's been very much consistent when I was running different experiments.